api/v1/models/Event.js

```diff
@@ -0,0 +1,93 @@
+/* jshint esversion 6 */
+var _Promise = require('bluebird');
+var _ = require('lodash');
+var moment = require('moment');
+
+var Model = require('./Model');
+var LocationEvents = require('./LocationEvents');
+
+var eventTags = require('../utils/eventTags.js');
+
+var Event = Model.extend({
+        tableName: 'events',
+        idAttribute: 'id',
+        hasTimestamps: ['created', 'updated'],
+        validations: {
+                name: ['required', 'string', 'maxLength:255'],
+                shortName: ['required', 'string', 'maxLength:25'],
+                description: ['required', 'string', 'maxLength:2047'],
+                qrCode: ['required', 'boolean'],
+                startTime: ['required', 'date'],
+                endTime: ['required', 'date'],
+                tag: ['required', 'string', eventTags.verifyTags]
+        },
+
+        locationEvents: function () {
+                return this.belongsToMany(LocationEvents);
+        }
+});
+
+/**
+ * Finds a event by its ID, joining in its related roles
+ * @param  {Number|String} id   the ID of the model with the appropriate type
+ * @return {Promise<Model>}     a Promise resolving to the resulting model or null
+ */
+Event.findById = function(id) {
```

Comment:
It doesn't look like you're fetching non-lazily though, so are you sure that's what the problem is?